### PR TITLE
Build spans within a context

### PR DESCRIPTION
**Pull request by dazuma**
Fixes https://github.com/census-instrumentation/opencensus-ruby/issues/16

### CODE REVIEW

File: test/trace/span_context_test.rb [hunk 1]
```diff
@@ -79,6 +79,52 @@
...
+    it "supports contains" do
+      root_context.contains?(context2).must_equal true
+      root_context.contains?(context1).must_equal true
+      root_context.contains?(root_context).must_equal true
+      context1.contains?(root_context).must_equal false
+      context2.contains?(context1).must_equal false
+    end
+
+    it "supports contained_span_builders" do
+      context2
```
Comment (chingor13) [inline test/trace/span_context_test.rb]: Is this needed?

File: lib/opencensus/trace/span_context.rb [hunk 2]
```diff
@@ -190,6 +228,22 @@ def this_span
...
+      # generated spans is undefined.
+      #
+      # Does not build any ancestor spans. If you want the entire span tree
+      # built, call this method on the `#root` context.
+      #
+      # @return [Array<Span>] Built Span objects
+      #
+      def build_contained_spans
+        contained_span_builders
+          .map { |sb| sb.finished? ? sb.to_span : nil }
```
Comment (chingor13) [inline lib/opencensus/trace/span_context.rb]: Map with a ternary and compact seems a bit odd. Maybe `filter` + `map` or `inject`?

File: lib/opencensus/trace/span_context.rb [hunk 3]
```diff
@@ -78,6 +87,35 @@ def root
...
+        false
+      end
+
+      ##
+      # Returns all SpanBuilder objects created by this context or any
+      # descendant context. The order of the returned spans is undefined.
+      #
+      # @return [Array<SpanBuilder>]
+      #
+      def contained_span_builders
```
Comment (chingor13) [inline lib/opencensus/trace/span_context.rb]: Does this need to be part of the public facing surface area? We could leave it public and annotate it to hide from docs if necessary.

Comment (dazuma) [reply lib/opencensus/trace/span_context.rb]: Made them private for now. I guess they're not needed yet.

File: lib/opencensus/trace/span_context.rb (hunk 2 above)
Comment (dazuma) [reply lib/opencensus/trace/span_context.rb]: Changed to find_all + map

File: test/trace/span_context_test.rb (hunk 1 above)
Comment (dazuma) [reply test/trace/span_context_test.rb]: Yes. Added a comment explaining why.

File: lib/opencensus/trace/span_context.rb [hunk 4]
```diff
@@ -287,6 +256,35 @@ def contained_span_builders
...
         end
       end
+
+      private
+
+      ##
+      # Create a child of this SpanContext, with a random unique span ID.
+      #
+      # @return [SpanContext] The created child context.
+      # @private
```
Comment (chingor13) [inline lib/opencensus/trace/span_context.rb]: nit: these don't need to be annotated private anymore if they are truly private.

Same for `get_span`

**Pull request merged**